yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -440,22 +454,43 @@ private[yarn] class YarnAllocator(
         // Hadoop 2.2.X added a ContainerExitStatus we should switch to use
         // there are some exit status' we shouldn't necessarily count against us, but for
         // now I think its ok as none of the containers are expected to exit
-        if (completedContainer.getExitStatus == ContainerExitStatus.PREEMPTED) {
-          logInfo("Container preempted: " + containerId)
-        } else if (completedContainer.getExitStatus == -103) { // vmem limit exceeded
-          logWarning(memLimitExceededLogMessage(
+        var isNormalExit = false
+        var containerExitReason = "Container exited for an unknown reason."
+        val exitStatus = completedContainer.getExitStatus
+        if (exitStatus == ContainerExitStatus.PREEMPTED) {
+          isNormalExit = true
```

Comment:
can you add a comment to explain why this is a normal exit, and link `SPARK-8167` in that comment?